File: electro-client/src/models/DocketVariantEliminatedResponse.ts
import BaseResponse from 'models/BaseResponse';
import { DocketReasonResponse } from 'models/DocketReason';
import { WarehouseResponse } from 'models/Warehouse';

export interface DocketVariantEliminatedResponse {
  docket: DocketResponse;
  quantity: number;
}

interface DocketResponse extends BaseResponse {
  type: number;
  code: string;
  reason: DocketReasonResponse;
  warehouse: WarehouseResponse;
  purchaseOrder: PurchaseOrderResponse | null;
  order: OrderResponse | null;
  status: number;
}

interface PurchaseOrderResponse extends BaseResponse {
  code: string;
  status: number;
}

interface OrderResponse extends BaseResponse {
  code: string;
  status: number;
}

File: electro-server/src/main/java/com/electro/dto/inventory/DocketVariantEliminatedResponse.java
package com.electro.dto.inventory;

import lombok.Data;
import org.springframework.lang.Nullable;

import java.time.Instant;

@Data
public class DocketVariantEliminatedResponse {
    private DocketVariantEliminatedResponse.DocketResponse docket;
    private Integer quantity;

    @Data
    public static class DocketResponse {
        private Long id;
        private Instant createdAt;
        private Instant updatedAt;
        private Integer type;
        private String code;
        private DocketReasonResponse reason;
        private WarehouseResponse warehouse;
        @Nullable
        private DocketVariantEliminatedResponse.DocketResponse.PurchaseOrderResponse purchaseOrder;
        @Nullable
        private DocketVariantEliminatedResponse.DocketResponse.OrderResponse order;
        private Integer status;

        @Data
        public static class PurchaseOrderResponse {
            private Long id;
            private Instant createdAt;
            private Instant updatedAt;
            private String code;
            private Integer status;
        }

        @Data
        public static class OrderResponse {
            private Long id;
            private Instant createdAt;
            private Instant updatedAt;
            private String code;
            private Integer status;
        }
    }
}
